File: ios/TodayInSpaceHistoryTests/Support/TestFixtures.swift
//
//  TestFixtures.swift
//  TodayInSpaceHistoryTests
//

import Foundation
@testable import TodayInSpaceHistory

enum TestFixtures {
    static func item(
        title: String? = "Title",
        description: String? = "Description",
        dateCreated: String?,
        href: String? = "https://example.com/asset.json"
    ) -> Item {
        let result = SearchResult(
            center: nil,
            dateCreated: dateCreated,
            description: description,
            keywords: nil,
            media_type: "image",
            nasaId: "id",
            title: title
        )
        return Item(data: [result], links: nil, href: href)
    }

    static func apiResponse(items: [Item]) -> APIResponse {
        APIResponse(
            collection: Collection(
                links: nil,
                href: nil,
                items: items
            )
        )
    }

    /// `date_created` for today's month/day in an arbitrary past year.
    static var todaysAnniversaryDateCreated: String {
        let calendar = Calendar(identifier: .gregorian)
        let today = Date()
        let month = calendar.component(.month, from: today)
        let day = calendar.component(.day, from: today)
        return String(format: "2001-%02d-%02dT12:00:00Z", month, day)
    }

    /// A `date_created` that is deliberately NOT today's month+day anniversary.
    /// Same month, day flipped to 1 (or 2 if today is already the 1st).
    static var nonAnniversaryDateCreated: String {
        let calendar = Calendar(identifier: .gregorian)
        let today = Date()
        let month = calendar.component(.month, from: today)
        let todayDay = calendar.component(.day, from: today)
        let day = todayDay == 1 ? 2 : 1
        return String(format: "2001-%02d-%02dT12:00:00Z", month, day)
    }
}

File: android/app/src/test/java/miko/todayinspacehistory/support/TestFixtures.kt
package miko.todayinspacehistory.support

import miko.todayinspacehistory.data.model.ApiResponse
import miko.todayinspacehistory.data.model.Collection
import miko.todayinspacehistory.data.model.Item
import miko.todayinspacehistory.data.model.SearchResult
import java.util.Calendar
import java.util.GregorianCalendar

object TestFixtures {
    const val ANNIVERSARY_DATE_CREATED = "1969-07-20T20:17:40Z"
    const val OTHER_DATE_CREATED = "1971-08-16T00:00:00Z"

    fun july20_2026(): Calendar {
        return GregorianCalendar().apply {
            clear()
            set(2026, Calendar.JULY, 20, 12, 0, 0)
        }
    }

    fun august9_2026(): Calendar {
        return GregorianCalendar().apply {
            clear()
            set(2026, Calendar.AUGUST, 9, 12, 0, 0)
        }
    }

    fun item(
        title: String? = "Title",
        description: String? = "Description",
        dateCreated: String?,
        href: String? = "https://example.com/asset.json",
    ): Item {
        val result = SearchResult(
            center = null,
            dateCreated = dateCreated,
            description = description,
            keywords = null,
            mediaType = "image",
            nasaId = "id",
            title = title,
        )
        return Item(data = listOf(result), links = null, href = href)
    }

    fun apiResponse(items: List<Item>): ApiResponse {
        return ApiResponse(
            collection = Collection(
                links = null,
                href = null,
                items = items,
            ),
        )
    }
}
